modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/notifications/data/NotificationsPreferencesRepositoryImpl.kt

```diff
@@ -0,0 +1,211 @@
+        return if (userActions.isEmpty()) context.getString(LR.string.none) else actionStrings
+    }
+
+    private fun getNotificationSoundSummary() = getRingtoneValue(settings.notificationSound.value.path)
+
+    private fun getRingtoneValue(ringtonePath: String?): String {
+        if (ringtonePath.isNullOrBlank()) {
+            return context.getString(LR.string.settings_notification_silent)
+        }
+        return when (val ringtone = RingtoneManager.getRingtone(context, ringtonePath.toUri())) {
+            null -> ""
+            else -> {
+                val title = ringtone.getTitle(context)
+                if (title == NotificationSound.defaultPath) {
+                    context.getString(LR.string.settings_notification_default_sound)
+                } else {
+                    title
+                }
+            }
+        }
+    }
+
+    override suspend fun setPreference(preference: NotificationPreference<*>) = withContext(dispatcher) {
+        when (preference.preference) {
+            NotificationPreferences.NEW_EPISODES_NOTIFY_ME -> {
+                val value = expectTypedValue<Boolean>(preference)
+                settings.notifyRefreshPodcast.set(value = value, updateModifiedAt = true)
+                podcastManager.updateAllShowNotifications(value)
+                if (value) {
+                    settings.setNotificationLastSeenToNow()
+                }
+            }
+
+            NotificationPreferences.SETTINGS_HIDE_NOTIFICATION_ON_PAUSE -> {
+                val value = expectTypedValue<Boolean>(preference)
+                settings.hideNotificationOnPause.set(value = value, updateModifiedAt = true)
+            }
+
+            NotificationPreferences.SETTINGS_PLAY_OVER -> {
+                val setting = expectTypedValue<PlayOverNotificationSetting>(preference)
+                settings.playOverNotification.set(value = setting, updateModifiedAt = true)
+            }
+
+            NotificationPreferences.NEW_EPISODES_VIBRATION -> {
+                val setting = expectTypedValue<NotificationVibrateSetting>(preference, fallbackToValue = NotificationVibrateSetting.DEFAULT)
+                settings.notificationVibrate.set(value = setting, updateModifiedAt = false)
+            }
+
+            NotificationPreferences.NEW_EPISODES_ACTIONS -> {
+                val setting = expectTypedValue<List<NewEpisodeNotificationAction>>(preference)
+                settings.newEpisodeNotificationActions.set(value = setting, updateModifiedAt = true)
+            }
+
+            NotificationPreferences.NEW_EPISODES_RINGTONE -> {
+                val setting = expectTypedValue<String>(preference)
+                settings.notificationSound.set(value = NotificationSound(setting, context), updateModifiedAt = false)
+            }
+
+            // Add handler for new preferences here
+            else -> Timber.d("Unhandled preference received in setPreference: ${preference.preference}")
```

Comment:
Same issue here as mentioned earlier with using `else` in a `when` statement. It makes it easy to miss new cases if we add more values later. Better to be explicit about all possible options.